src/s2wasm.h

```diff
@@ -465,7 +466,17 @@ class S2WasmBuilder {
   }
 
   void parseGlobl() {
-    (void)getStr();
+    auto str = getStr();
+
+    globls.insert(str);
+
+    if (wasm.functionsMap.count(str)) {
```

Comment:
instead of doing this here and in the other place below, how about doing a loop over `globls` in `fix()`, which runs at the very end?